### PR TITLE
CI: add Canadian Cross build job

### PR DESCRIPTION
**Pull request by temap**
Add reusable GitHub workflow to build Canadian Cross toolchain where
the x86_64-w64-mingw32 cross-compiler is used which is built with ct-ng.

Signed-off-by: Artem Panfilov <artemp@synopsys.com>

### CODE REVIEW

File: .github/workflows/continuous-integration-workflow.yml [hunk 1]
```diff
@@ -9,10 +9,7 @@ jobs:
     runs-on: ${{ matrix.host }}
     strategy:
       matrix:
-        host: [
-          "ubuntu-latest",
-          "macos-10.15",
-        ]
+        host: ["ubuntu-latest", "macos-10.15"]
```
Comment (abrodkin) [inline .github/workflows/continuous-integration-workflow.yml:12]: I'd move that beautification to a separate commit to not pollute real meat of the functional change.

File: .github/workflows/continuous-integration-workflow.yml [hunk 2]
```diff
@@ -54,9 +51,7 @@ jobs:
     runs-on: ${{ matrix.host }}
     strategy:
       matrix:
-        host: [
-          "ubuntu-latest",
-        ]
+        host: ["ubuntu-latest"]
```
Comment (abrodkin) [inline .github/workflows/continuous-integration-workflow.yml:54]: Ditto

File: .github/workflows/continuous-integration-workflow.yml [hunk 3]
```diff
@@ -122,71 +113,16 @@ jobs:
           "s390-unknown-linux-gnu",
           "sh-unknown-elf",
           "sparc-unknown-linux-gnu",
-          "x86_64-unknown-linux-gnu",
           "x86_64-multilib-linux-uclibc",
+          "x86_64-unknown-linux-gnu",
```
Comment (abrodkin) [inline .github/workflows/continuous-integration-workflow.yml:117]: Ditto

**Review by abrodkin — APPROVED** [review]
I'd say it's a long awaited feature as we discussed that previously - testing of canadian cross tools is good to have. And now when it's possible to re-use parts of the existing flow, implementation is very simple and clean.

The only concern I have it's a limitation of GitHub CI for non-paid accounts so that we cannot see this check in action - lots of jobs are pending still after many hours. But w/o funding what else can we do?

**Comment by cpackham** [comment]
I certainly like the look of this. We might be able to leverage azure, circle-ci, gitlab-ci to spread the load a bit. I am surprised no-one from GitHub has complained about us chewing up resources.

**Comment by abrodkin** [comment]
> I am surprised no-one from GitHub has complained about us chewing up resources.

Well I guess the answer is simple - we consume all the given for free resources and then we're being put on halt. So they don't see resources consumption higher that they expect or allow us to use. But given what we do here is a true open source development and results of our works are being used in some other projects, maybe it worth contacting GitHub folks asking for more resources?

File: .github/workflows/continuous-integration-workflow.yml (hunk 1 above)
Comment (temap) [reply .github/workflows/continuous-integration-workflow.yml:12]: OK, I will revert code changes, not related to PR.

Comment (abrodkin) [reply .github/workflows/continuous-integration-workflow.yml:12]: There's no need to revert, just move them to a separate patch to be applied before the bigger functional change in the series.

**Comment by temap** [comment]
@cpackham, all checks passed. Do you have any concerns related to this PR except @abrodkin comments?

**Comment by cpackham** [comment]
No concerns from me. If you update to address @abrodkin's points I'll take
a look this weekend.

**Review by cpackham — APPROVED** [review]
